### PR TITLE
fix Dropout's bug

### PR DESCRIPTION
**Pull request by fmscole**

### CODE REVIEW

**Review by froystig — CHANGES_REQUESTED** [review]
Thanks for the PR. After we address the comment above, could you also [squash your commits](https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests)?

**Comment by fmscole** [comment]
I'm sorry.
I have squshed my commit.

I don't think it's a good choice to keep the default parameters here. Because many people may not pay attention to this and lead to potential errors, and they may not receive any prompts.




温暖的晚风
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "google/jax"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月3日(星期六) 上午8:54
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [google/jax] fix Dropout's bug (PR #16229)





 
@froystig requested changes on this pull request.
 
Thanks for the PR. After we address the comment above, could you also squash your commits?

 
In jax/example_libraries/stax.py:
 &gt;    """Layer construction function for a dropout layer with given rate."""    def init_fun(rng, input_shape):      return input_shape, ()    def apply_fun(params, inputs, **kwargs): -    rng = kwargs.get('rng', None) -    if rng is None: -      msg = ("Dropout layer requires apply_fun to be called with a PRNG key " -             "argument. That is, instead of `apply_fun(params, inputs)`, call " -             "it like `apply_fun(params, inputs, rng)` where `rng` is a " -             "jax.random.PRNGKey value.") +    mode = kwargs.get('mode', None)  
Is this change needed as well, or can we keep the mode argument the way it was, as an optional keyword argument with default value 'train' (as in the code snippet that you shared in #16218)?

**Comment by froystig** [comment]
It would be better to keep the `mode` argument the way it was, so that the function's signature conveys its parameters. Stax is not meant to have any users, but instead to be a succinct example library.

Also, it looks like squashing did not work out. This PR now has 6 commits and conflicts with the main branch. Could you try again?
